src/main/java/graphql/incremental/IncrementalExecutionResultImpl.java

```diff
@@ -52,6 +53,13 @@ public static Builder fromExecutionResult(ExecutionResult executionResult) {
         return new Builder().from(executionResult);
     }
 
+    @Override
+    public IncrementalExecutionResult transform(Consumer<ExecutionResult.Builder<?>> builderConsumer) {
```

Comment:
It would be nice if this Consumer was actually on type `IncrementalExecutionResult.Builder`, but changing it would either break the override, or would require a lot of generics to be added to ExecutionResult.

As written, the builder given to the consumer can be cast to `IncrementalExecutionResult.Builder`. I'm open to feedback/ideas on if there's a better way to have nicer types.

Reply:
Good point.
That's a tricky problem though, and I can't think of a nice solution right now.
I'm happy to merge the PR as is if it unblocks you.